app/src/main/java/com/example/next_show/data/TraktApplication.java

```diff
@@ -32,4 +46,29 @@ public Shows getNewShowsInstance(){
         Log.i(TAG, "Created new instance for Trakt");
         return traktShows;
     }
+
+    public static void fetchTrendingShows(Shows showsObj, FeedFragment.ShowCallback showCallback) {
+        Log.i(TAG, "Fetching shows now!");
+        try {
+            // enqueue to do asynchronous call and execute to do it synchronously
+            showsObj.trending(PAGES_REQUESTED, LIMIT, Extended.FULL).enqueue(new Callback<List<TrendingShow>>() {
+                @Override
+                public void onResponse(Call<List<TrendingShow>> call, Response<List<TrendingShow>> response) {
+                    if (response.isSuccessful()) {
+                        showCallback.onSuccess(Show.formatTrendingShows(response.body()));
+                    } else {
+                        showCallback.onFailure(response.code());
+                    }
+                }
+
+                @Override
+                public void onFailure(Call<List<TrendingShow>> call, Throwable t) {
+                    Log.e(TAG, "OnFailure", t);
```

Comment:
You might also want to handle the error case here.  If you fail here, will the feed fragment ever know that it failed?